gapic-common/lib/gapic/common/retry_policy.rb

```diff
@@ -0,0 +1,196 @@
+      DEFAULT_INITIAL_DELAY = 1
+      DEFAULT_MAX_DELAY = 15
+      DEFAULT_MULTIPLIER = 1.3
+      DEFAULT_RETRY_CODES = [].freeze
+      DEFAULT_TIMEOUT = 3600 # One hour
+
+      ##
+      # Create new Gapic::Common:RetryPolicy instance.
+      #
+      # @param initial_delay [Numeric] Initial delay in seconds for this policy.
+      # @param max_delay [Numeric] Maximum delay in seconds for this policy.
+      # @param multiplier [Numeric] The delay scaling factor for each subsequent retry attempt.
+      # @param retry_codes [Array<String|Numeric>] List of retry codes.
+      # @param timeout [Numeric] Timeout threshold value in seconds.
+      #
+      def initialize initial_delay: nil, max_delay: nil, multiplier: nil, retry_codes: nil, timeout: nil
+        instance_variable_set "@initial_delay", initial_delay
+        instance_variable_set "@max_delay", max_delay
+        instance_variable_set "@multiplier", multiplier
+        instance_variable_set "@retry_codes", convert_codes(retry_codes)
+        instance_variable_set "@timeout", timeout
+        instance_variable_set "@retry_block", nil
+        @delay = nil
+      end
+
+      def initial_delay
+        @initial_delay || DEFAULT_INITIAL_DELAY
+      end
+
+      def max_delay
+        @max_delay || DEFAULT_MAX_DELAY
+      end
+
+      def multiplier
+        @multiplier || DEFAULT_MULTIPLIER
+      end
+
+      def retry_codes
+        @retry_codes || DEFAULT_RETRY_CODES
+      end
+
+      def timeout
+        @timeout || DEFAULT_TIMEOUT
+      end
+
+      ##
+      # Customize retry logic based on keyword arguments.
+      #
+      # @param [Block] Custom block yielding whether this policy is retriable based on keyword arguments.
+      def customize_retry &retry_block
+        if retry_block.nil?
+          raise ArgumentError, "A block must be provided for custom retry logic."
+        end
+        @retry_block = retry_block
+
+        self
+      end
+
+      ##
+      # Perform delay if and only if retriable.
+      #
+      # Positional argument `error` takes precedence over keyword arguments, meaning the latter
+      # will only be used iff `error` is not present.
+      #
+      # @return [Boolean] Whether the delay was executed.
+      #
+      def call error = nil, **kwargs
+        should_retry = error.nil? ? retry?(**kwargs) : retry_error?(error)
+        return false unless should_retry
+        perform_delay!
+      end
+      alias perform_delay call
+
+      ##
+      # Perform delay.
+      #
+      # @return [Boolean] Whether the delay was executed.
+      #
+      def perform_delay!
+        delay!
+        increment_delay!
+        true
+      end
+
+      ##
+      # Current delay value in seconds.
+      #
+      # @return [Numeric] Time delay in seconds.
+      #
+      def delay
+        @delay || initial_delay
+      end
+
+      ##
+      # @private
+      # @return [Boolean] Whether this error should be retried.
+      #
+      def retry_error? error
```

Comment:
If this is not intended to be part of the public interface, maybe just put it below the `private` keyword.

Reply:
`retry_error?` is being used by the polling harness in `wait`, so keeping as is.